### PR TITLE
SQL initial commit

### PR DESCRIPTION
**Pull request by bestdark**
+ Table users
+ Register User func
+ Sign In func
+ Check if signed func
+ tests

### CODE REVIEW

File: SQL/create.sql [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+	IN Email VARCHAR(50),
+	IN Pass VARCHAR(50),
+	OUT SessionExpire DATETIME,
+	OUT SessionToken binary(64)
+)
+BEGIN
+	IF
+	(SELECT 1 FROM Users	WHERE Users.Email = Email AND Users.PassHash = (@Hash:=binary( SHA2 ( CONCAT ( CONVERT ( Users.Salt , char(11) ) ,Pass ),256))))
+	THEN
+		SELECT ADDTIME(NOW(), '14 00:00:00') INTO SessionExpire;
```
Comment (ALEXKIRNAS) [inline SQL/create.sql]: Let's create some constant with explicit name.

File: SQL/create.sql [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+		SELECT SHA2 (CONCAT(SessionExpire,@Hash),0) INTO SessionToken;
+	END IF;
+END
+//
+DELIMITER ; 
+
+CALL signIn('admin','qwerty',@a,@b);
```
Comment (ALEXKIRNAS) [inline SQL/create.sql]: Create new file test_create.sql and put this code there.

File: SQL/create.sql [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+BEGIN
+	IF ((SELECT 1 
+		 FROM Users 
+		 WHERE Users.Email=Email AND
+		       SHA2(CONCAT(SessionExpire,Users.PassHash),0)=SessionToken)
+		AND
+		SessionExpire >= NOW()
+		)
+	THEN
+	RETURN 1;
```
Comment (ALEXKIRNAS) [inline SQL/create.sql]: Why you specify return type as BOOL and return INT?

File: SQL/create.sql [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+	ELSE
+	RETURN 0;
+	END IF;
+END
+//
+DELIMITER ; 
+SELECT checkSession('admin',@a,@b);
```
Comment (ALEXKIRNAS) [inline SQL/create.sql]: Create new file test_create.sql and put this code there.

File: SQL/create.sql [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+	SET @Rnd = FLOOR(RAND()*2147483647);
+	INSERT INTO Users(Users.Email,Users.Salt,Users.PassHash) Values 
+		(Email,
+		@Rnd,
+		binary( SHA2 ( CONCAT ( CONVERT ( @Rnd , char(11) ) ,Pass ),256)));
+END
+//
+DELIMITER ; 
+
+CALL regUser('admin','DemonicSwordZariche');
```
Comment (ALEXKIRNAS) [inline SQL/create.sql]: Can you separate code and test? Create new file test_create.sql and put this code there.

File: SQL/create.sql (hunk 3 above)
Comment (bestdark) [reply SQL/create.sql]: These types are synonyms for TINYINT(1)
https://mariadb.com/kb/en/library/boolean/

**Pull request merged**